src/DotLiquid/StandardFilters.cs

```diff
@@ -109,22 +109,15 @@ public static string UrlDecode(string input)
         }
 
         /// <summary>
-        /// capitalize words in the input sentence
+        /// Capitalizes the first word in each sentence of <paramref name="input"/>
         /// </summary>
         /// <param name="input"></param>
         /// <returns></returns>
         public static string Capitalize(string input)
         {
-            if (input.IsNullOrWhiteSpace())
-                return input;
-
-            return string.IsNullOrEmpty(input)
+            return input.IsNullOrWhiteSpace()
                 ? input
-#if CORE
-                : Regex.Replace(input, @"\b(\w)", m => m.Value.ToUpper(), RegexOptions.None, Template.RegexTimeOut);
-#else
-                : CultureInfo.CurrentCulture.TextInfo.ToTitleCase(input);
-#endif
+                : Regex.Replace(input.ToLower(), @"^\D|\.( ){1,}\D", m => m.Value.ToUpper(), RegexOptions.None, Template.RegexTimeOut);

```

Comment:
I don't know that ToLower, i.e. everything else forced to lower case is the intent of the spec. The spec says "so later words are not affected"

The Shopify spec example does not include a capital letter in the input so it is difficult to say for sure.

My guess is that the intent is to leave other characters unchanged. If the input had a proper noun, setting all words in the sentence to lower case may not be desirable.

Either avoid ToLower or get a clarification from Shopify by asking their engineers what is the intended behavior (e.g. via their forum or their presence on twitter).

Reply:
Looks like the shopify implementation is using String.capitalize, which does lower everything else. And also doesn't transform chars after a dot or an hyphen, only spaces.

https://github.com/Shopify/liquid/blob/a03f02789b14b0728e68c800c6699292a4c76672/lib/liquid/standardfilters.rb#L40

`"HellO. wOrld"` => `"Hello. world"`

https://repl.it/repls/MeanHumbleVolcano

Reply:
The code speaks, I guess. Shopify might not have documented by we can run their code to see what they implemented. Funny that the very line in their code is preceded by "capitalize words in the input centence" [sic]